src/IssueViz.UnitTests/Editor/QuickActions/QuickFixes/QuickFixActionsSourceTests.cs

```diff
@@ -93,14 +94,108 @@ public void OnTagsChanged_DismissLightBulbSession()
             var issueLocationsTagAggregator = new Mock<ITagAggregator<IIssueLocationTag>>();
             var lightBulbBroker = new Mock<ILightBulbBroker>();
 
-            CreateTestSubject(issueLocationsTagAggregator.Object, lightBulbBroker: lightBulbBroker.Object);
+            CreateTestSubject(issueLocationsTagAggregator.Object, lightBulbBroker.Object);
 
             lightBulbBroker.VerifyNoOtherCalls();
 
             issueLocationsTagAggregator.Raise(x => x.TagsChanged += null, new TagsChangedEventArgs(Mock.Of<IMappingSpan>()));
             lightBulbBroker.Verify(x => x.DismissSession(textView), Times.Once);
         }
 
+        [TestMethod]
+        public void OnTagsChanged_NonCriticalException_ExceptionIsCaught()
+        {
+            var eventHandle = new EventWaitHandle(false, EventResetMode.ManualReset);
+
+            var threadHandling = new Mock<IThreadHandling>();
+            threadHandling
+                .Setup(x => x.RunOnUIThread(It.IsAny<Action>()))
+                .Callback((Action callbackAction) =>
+                {
+                    try
+                    {
+                        callbackAction();
+                    }
+                    finally
+                    {
+                        eventHandle.Set(); // signal the test that the action has finished
```

Comment:
The event itself is void, but the implementation is async. There is no way to wait for the event to finish as that goes against the fire-and-forget architecture of events.
This was the solution I came up with and it's not ideal.